### ISSUE TITLE
Index out of bound when joining dataframe with nullable columns

### ISSUE THREAD

**Issue by potter420** [pola-rs/polars#1077]
#### Are you using Python or Rust?

**Python**.

#### What version of polars are you using?

0.8.15

#### What operating system are you using polars on?

Amazon Linux 2

#### Describe your bug.

When we join a dataframe of size 1 and a dataframe with null column

#### What are the steps to reproduce the behavior?

```python
import polars as pl

a = pl.DataFrame({
    'a':[1],
    'b':[None]
})
b = pl.DataFrame({
    'a':[1,2,3,4],
    'c':[1,None,3,4]
})
a.join(b, on='a')
```

#### What is the actual behavior?
```
PanicException: index out of bounds: the len is 0 but the index is 0
```

Furthermore, In the one other case:
```python
a = pl.DataFrame({
    'a':[1],
    'b':[1]
})
b = pl.DataFrame({
    'a':[1,2,3,4],
    'c':[1,None,3,4]
})
a.join(b, on='a')
```
```
PanicException: assertion failed: ceil(offset + len, 8) <= buffer.len() * 8
```

#### What is the expected behavior?

It should have join successfully and produce expected output:
```
shape: (1, 3)
╭─────┬──────┬─────╮
│ a   ┆ b    ┆ c   │
│ --- ┆ ---  ┆ --- │
│ i64 ┆ f64  ┆ i64 │
╞═════╪══════╪═════╡
│ 1   ┆ null ┆ 1   │
╰─────┴──────┴─────╯
```

**Comment by ritchie46** [comment]
Thanks @potter420, this is fixed in #1079

**Comment by potter420** [comment]
Thanks you very much @ritchie46
